app/controllers/merge_requests_controller.rb

```diff
@@ -63,19 +73,30 @@ def merge_request_params
   end
 
   def merge_request_organisation_params
-    { merge_request: @merge_request, merging_organisation_id: params[:merge_request][:merging_organisation] }
+    {
+      merge_request: @merge_request,
+      merging_organisation_id: params.dig(:merge_request, :merging_organisation),
+    }
   end
 
   def find_resource
     @merge_request = MergeRequest.find(params[:id])
   end
 
   def next_page_path
-    absorbing_organisation_merge_request_path(@merge_request)
+    if params.dig(:merge_request, :absorbing_organisation_id).present?
+      confirm_telephone_number_merge_request_path(@merge_request)
+    else
+      absorbing_organisation_merge_request_path(@merge_request)
+    end
```

Comment:
Hmm, maybe it's worth passing through a page/current_page in params? Similar to what we do in schemes